File: backend/DataAccess/Models/TrackedItem.cs

namespace backend.DataAccess.Models;

public class TrackedItem
{

    /// <summary>
    /// Primary key of the tracked item.
    /// </summary>
    public int Id { get; set; }

    /// <summary>
    /// Title of the tracked item.
    /// </summary>
    public required string Title { get; set; }

    /// <summary>
    /// Creator of the tracked item.
    /// </summary>
    public required Guid UserId { get; set; }

    /// <summary>
    /// Entries of the tracked item.
    /// </summary>
    public List<TrackingEntry> Entries { get; set; } = [];
}

File: frontend/src/model/trackedItem.ts
import type { TrackingEntry } from './trackingEntry'

export class TrackedItem {
  id: number
  title: string
  trackingEntries: TrackingEntry

  constructor(id: number, title: string, trackingEntries: TrackingEntry) {
    this.id = id
    this.title = title
    this.trackingEntries = trackingEntries
  }
}
